docs/src/pages/afm.md

```diff
@@ -0,0 +1,228 @@
+
+The Anywidget Front-end Module is widget front-end code authored by a widget
+developer. It contains the front-end logic of a widget, which is defined by
+implementing various [lifecycle methods](#lifecycle-methods) to control the
+widget's behavior. AFM is a web-standard ECMAScript module (ESM) can be
+authored simply as a text file or generated from a more complex front-end
+toolchain.
+
+### Host platform
+
+The web-based environment in which a widget is embedded. It is responsible for
+loading AFM and calling [lifecycle methods](#lifecycle-methods) with required
+the platform APIs.
+
+The `anywidget` Python library provides the necessary glue code to make all
+Jupyter-like environments (e.g., Jupyter Notebook, JupyterLab, Google Colab, VS
+Code) an AFM-compatible host platform. The
+[marimo](https://github.com/marimo-team/marimo) project is a _native_ host
+platform.
+
+## Anywidget Front-end Module (AFM)
+
+An Anywidget Front-End Module (AFM) is an [ECMAScript
+module](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)
+that defines a widget's behavior through [lifecycle
+methods](#lifecycle-methods).
+
+```js
+export default {
+  initialize({ model }) {
+    // Set up shared state or event handlers.
+    return () => {
+      // Optional: Called when the widget is destroyed.
+    } 
+  },
+  render({ model, el }) {
+    // Render the widget's view into the el HTMLElement.
+    return () => {
+      // Optional: Called when the view is destroyed.
+    }
+  }
+}
+```
+
+A host platform is expected to:
+
+- Load this module.
+- Call the lifecycle methods, passing in dependencies (`model` & `el`).
+
+All browsers support ESM, so loading the module is supported natively across
+web-based environments. It is then the host platform's responsibility to
+implement the required [`model` interface](#model-interface) and provide an
+output DOM element (`el`). This simple mechanism allows new host platforms to
+be implemented as long as they adhere to these requirements.
+
+### Lifecycle Methods
+
+The widget lifecycle in AFM follows a Model-View pattern, consisting of two
+main phases:
+
+- **Model Initialization**: Occurs when a widget is first created, setting up
+the model and any shared state.
+- **View Rendering**: Happens each time a widget needs to be displayed,
+potentially multiple times for a single widget instance.
+
+AFM exports methods that correspond to these lifecycle phases. The default
+export object specifies one or more widget lifecycle hooks:
+
+- `initialize({ model })`: Executed once per widget instance during model
+initialization. It has access to `model` to setup non-view event handlers or
+state to share across views.
+
+- `render({ model, el })`: Executed once per view during view rendering. It has
+access to both the `model` and an `el` DOM element. Used to setup event
+handlers or access state specific to that view.
+
+Each method MAY return a **cleanup function** which will be called when the
```

Comment:
Does the cleanup function have a signature (take optional arguments)?

Reply:
Nope, just a () => void. You can use a closure to capture any deps.